sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/plans/logical/EventTimeWatermark.scala

```diff
@@ -28,9 +27,7 @@ object EventTimeWatermark {
   val delayKey = "spark.watermarkDelayMs"
 
   def getDelayMs(delay: CalendarInterval): Long = {
-    // We define month as `31 days` to simplify calculation.
-    val millisPerMonth = TimeUnit.MICROSECONDS.toMillis(CalendarInterval.MICROS_PER_DAY) * 31
-    delay.milliseconds + delay.months * millisPerMonth
+    delay.milliseconds + delay.months * MILLIS_PER_MONTH
```

Comment:
What we need here is seconds per month, not seconds per year. I think we should still assume 31 days per month here.

Reply:
Can you point out where we need seconds/days per year in the codebase?

Reply:
I believe any place including this one when we need a duration (in seconds or its fractions). The difference between `months_between()` and this place is `months_between` uses month length to calculate fraction of month, and 28 or 31 days per months don't really matter because it impacts on 2nd or 3rd digit in fractions but here we operate on bigger numbers when months form years. And it become matter how much days we use per year. Let's say we calculate duration of 10 years which 120 months. If we use 31 days per months, this duration is 31 * 120 = 10 * 372 = 3720 days but if one year is 365.2425, than 1 year = 3652. The difference is 3720 - 3652 = 68 days or the calculation error is more than 2 months. That's matter I believe.

Reply:
`months_between` is sort of a special case because "31 days per month" is (it seems) actually how it is supposed to work, correctly.

It's rare that someone would specify "1 month" here, let alone "10 years" right? or am I missing something? these are things like watermark intervals. Not that it means the semantics don't matter, it's just quite a corner case.

I therefore just don't feel strongly either way about it. We don't need to match `months_between` semantics. More precision is nice, but surely it almost never comes up anyway? I don't mind the change, as a result.